service/controller/chart/resource/status/create.go

```diff
@@ -100,9 +101,27 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
+func (r *Resource) getAuthToken(ctx context.Context) (string, error) {
+	secret, err := r.k8sClient.CoreV1().Secrets(namespace).Get(ctx, authTokenName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// it is certainly control plane apps or auth secret is not created yet.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no auth token secret yet in cluster")
```

Comment:
```suggestion
		// There is no auth token secret. It may not have been created yet.
		r.logger.LogCtx(ctx, "level", "debug", "message", "no auth token secret found skip calling webhook")
```

Good to log this but can you also check if the app CR uses InCluster? If so we can skip this API call.

Reply:
Do you mean `chart` CRs? In chart-operator, I'm not sure how to check `InCluster` for chart CRs.

Reply:
Thanks you're right I was mixing up the CRs. There is no good way to detect that in the chart CR so this is fine, I updated my suggestion for the comment. WDYT?

Reply:
Perfect, thanks for your suggestion, I'm going to change that.